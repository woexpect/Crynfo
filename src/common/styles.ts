import {Dimensions, Platform} from 'react-native';
import {ScaledSheet, verticalScale} from 'react-native-size-matters';

const {width, height} = Dimensions.get('window');

const isIosDevice = Platform.OS === 'ios';

export default ScaledSheet.create({
  // Layout Styles
  completeScreen: {
    flex: 1,
  },
  centerAll: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  completeWidth: {
    width: '100%',
  },
  cryptoListContainer: {
    width: '100%',
    height: verticalScale(64),
  },
  hideOverflow: {
    overflow: 'hidden',
  },
  centerVertically: {
    justifyContent: 'center',
  },
  spaceEvenly: {
    justifyContent: 'space-evenly',
  },
  spaceBetween: {
    justifyContent: 'space-between',
  },
  toEnd: {
    alignItems: 'flex-end',
  },
  roundedTopCorners: {
    borderTopRightRadius: verticalScale(16),
    borderTopLeftRadius: verticalScale(16),
  },
  onTop: {
    position: 'absolute',
  },
  cover: {
    width: '100%',
    height: '100%',
  },
  row: {
    flexDirection: 'row',
  },
  // Paddings
  paddingRight16: {
    paddingRight: verticalScale(16),
  },
  paddingHorizontal: {
    paddingHorizontal: verticalScale(16),
  },
  // Margins
  marginLeft16: {
    marginLeft: verticalScale(16),
  },
  marginRight16: {
    marginRight: verticalScale(16),
  },
  marginTop8: {
    marginTop: verticalScale(8),
  },
  marginTop1: {
    marginTop: 1,
  },
  marginTop16: {
    marginTop: verticalScale(16),
  },
  marginTop24: {
    marginTop: verticalScale(24),
  },
  // Background Colors
  bgBlack: {
    backgroundColor: '#141414',
  },
  bgMain: {
    backgroundColor: '#DEF263',
  },
  // Tint Colors
  tintBlack: {
    tintColor: '#141414',
  },
  tintMain: {
    tintColor: '#DEF263',
  },
  // Text Colors
  tcRed: {
    color: '#F26363',
  },
  tcBlack: {
    color: '#141414',
  },
  tcMain: {
    color: '#DEF263',
  },
  tcBlack8: {
    color: '#1414141F',
  },
  tcBlack64: {
    color: '#141414A3',
  },
  tcMain16: {
    color: '#DEF26329',
  },
  // Border Colors
  bcBlack: {
    borderColor: '#141414',
  },
  bcMain: {
    borderColor: '#DEF263',
  },
  // Font Families
  black: {
    fontFamily: 'Inter-Black',
  },
  bold: {
    fontFamily: 'Inter-Bold',
  },
  semiBold: {
    fontFamily: 'Inter-SemiBold',
  },
  medium: {
    fontFamily: 'Inter-Medium',
  },
  regular: {
    fontFamily: 'Inter-Regular',
  },
  // Font sizes
  cryptoPlaceholder: {
    fontSize: '112@ms0.3',
    includeFontPadding: false,
    lineHeight: '112@ms0.3',
    marginTop: -8,
    marginLeft: -4,
    letterSpacing: -2,
  },
  title: {
    fontSize: '26@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h1: {
    fontSize: '24@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h15: {
    fontSize: '22@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  h2: {
    fontSize: '20@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  subtitle: {
    fontSize: '18@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largerText: {
    fontSize: '16@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  largeText: {
    fontSize: '14@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  mediumText: {
    fontSize: '12@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  normalText: {
    fontSize: '10@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
  smallText: {
    fontSize: '6@ms0.3',
    includeFontPadding: false,
    letterSpacing: 0,
  },
});
