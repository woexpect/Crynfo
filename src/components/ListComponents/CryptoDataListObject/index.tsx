import React from 'react';
import {Text, View} from 'react-native';
// Custom components
// User tools
import styles from '../../../common/styles';

const CryptoDataListObject = ({
  addStyles,
  label = '',
  value = '',
  negativeValue = false,
}: {
  addStyles?: any;
  label: string;
  value: string;
  negativeValue?: boolean;
}) => {
  return (
    <View
      style={[
        styles.completeWidth,
        styles.row,
        styles.spaceBetween,
        styles.paddingHorizontal,
        addStyles,
      ]}>
      <Text style={[styles.regular, styles.tcMain]}>{label}</Text>
      <Text
        style={[styles.regular, styles.tcMain, negativeValue && styles.tcRed]}>
        {value}
      </Text>
    </View>
  );
};

export default React.memo(CryptoDataListObject);
