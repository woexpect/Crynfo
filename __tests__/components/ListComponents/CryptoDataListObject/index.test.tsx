import 'react-native';
import React from 'react';
import CryptoDataListObject from '../../../../src/components/ListComponents/CryptoDataListObject';

// Note: import explicitly to use the types shiped with jest.
import {expect, it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import styles from '../../../../src/common/styles';

it('renders correctly with no values', () => {
  const tree = renderer.create(<CryptoDataListObject />);

  expect(tree).toMatchSnapshot();
});

it('renders correctly with required values only', () => {
  const tree = renderer.create(
    <CryptoDataListObject label="Test Crypto" value="TC" />,
  );

  expect(tree).toMatchSnapshot();
});

it('renders correctly with all possible values', () => {
  const tree = renderer.create(
    <CryptoDataListObject
      addStyles={styles.marginTop24}
      label="Test Crypto2"
      value="TC2"
      quote=""
      negativeValue
      noFloat
    />,
  );

  expect(tree).toMatchSnapshot();
});
