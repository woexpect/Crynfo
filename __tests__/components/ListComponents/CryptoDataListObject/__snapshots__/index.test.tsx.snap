// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all possible values 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "marginTop": 47.08235294117647,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-Regular",
        },
        {
          "color": "#DEF263",
        },
      ]
    }
  >
    Test Crypto2
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          {
            "color": "#F26363",
          },
        ]
      }
    >
      TC2
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          {
            "marginLeft": 31.388235294117646,
          },
          {
            "color": "#F26363",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly with no values 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-Regular",
        },
        {
          "color": "#DEF263",
        },
      ]
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          false,
        ]
      }
    >
      $NaN
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          {
            "marginLeft": 31.388235294117646,
          },
          false,
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly with required values only 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Inter-Regular",
        },
        {
          "color": "#DEF263",
        },
      ]
    }
  >
    Test Crypto
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          false,
        ]
      }
    >
      $NaN
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "Inter-Regular",
          },
          {
            "color": "#DEF263",
          },
          {
            "marginLeft": 31.388235294117646,
          },
          false,
        ]
      }
    />
  </View>
</View>
`;
